jsx/Modal.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * This file contains the React Component for a Modal Window.
+ *
+ * @author Henri Rabalais
+ * @version 1.1.0
+ *
+ */
+import swal from 'sweetalert2';
+
+/**
+ * Modal Component.
+ * React wrapper for a Modal Window. Allows to dynamically toggle a Modal window.
+ *
+ * ================================================
+ * Usage:
+ * - Wrap the contents to be displayed by the Modal Window by the Modal Component.
+ * - Use the 'show' prop to toggle the Modal Component's presentation.
+ * - Use the 'title' prop to set a title for the Modal Component.
+ * - Use the 'onSubmit' prop set a submission function for the Modal's contents.
+ * - Use the 'throwWarning' prop to throw a warning upon closure of the Modal Window
```

Comment:
looking to test. could you possibly add the usage for the closeModal prop? and could adding a default for that prop be useful?

Reply:
For sure! I'll add it in there. Basically the closeModal props is the function that is triggered when the modal is closed. This should a be a function within the parent which changes a state so that the value passed as `show` becomes false. It's a required function, so I don't think it should have a default per se, but I could throw a console warning if it is not set. I had been planning on adding an implementation of the modal in loris somewhere so it would be easier to test. I can do that in my next commit.

Reply:
what if there is no function that should be triggered i.e. if I click close and all i want is the modal to close and display the window behind it? I'm finding it hard to understand what this required closeModal function should be

Reply:
If there is no function that is triggered, then the modal will not closed. The value of the prop `show` needs to go from `true` to `false` for the modal window is disappear. The function would look something like this.

```
closeModal() {
  let showModal = this.state.showModal;
  showModal = false;
  this.setState({showModal});
}
```

`this.state.showModal` would then be passed to `Modal` as the prop `show`.